sherpa/models/tests/test_model.py

```diff
@@ -149,11 +158,77 @@ def test_complex_expression(self):
 
     def test_filter(self):
         m = self.s[::2]
-        self.assertTrue(type(m) is FilterModel)
+        self.assertIs(type(m), FilterModel)
         self.assertTrue(numpy.all(m(self.xx) == self.s(self.xx)[::2]))
 
     def test_nested(self):
         for func in (numpy.sin, self.s):
             m = self.m.apply(func)
-            self.assertTrue(type(m) is NestedModel)
+            self.assertIs(type(m), NestedModel)
             self.assertEqual(m(self.x), func(self.m(self.x)))
+
+
+# Test support for renamed parameters by sub-classing
+# the Sin model (which lets the tests be re-used).
+#
+class RenamedPars(Sin):
+
+    def __init__(self, name='renamedpars'):
+        self._renamedpars = [('norm', 'ampl')]
+        Sin.__init__(self, name)
+
+
+class test_model_renamed(test_model):
+
+    def setUp(self):
+        self.m = RenamedPars('m')
+
+    def test_getpar_rename(self):
+        for par in (self.m.norm, self.m.NorM, self.m.NOrm):
+            self.assertIs(par, self.m.pars[2])
+
+    def test_setpar_rename(self):
+        self.m.ampl = 1
+        self.assertNotEqual(self.m.ampl.val, 12.0)
+        self.m.norm = 12
+        self.assertEqual(self.m.ampl.val, 12.0)
+        self.m.NoRM = 18
+        self.assertEqual(self.m.ampl.val, 18.0)
+        self.m.ampl = 1
+        self.assertEqual(self.m.ampl.val, 1.0)
+
+
+# During testing an XSPEC model was found to refer to one of its
+# parameters with a different case to how it was created, which
+# caused problems. Replicate this problem case here.
+#
+class ParameterCase(ArithmeticModel):
+    """Re-implemenent Sin model so can copy tests"""
+
+    def __init__(self, name='parametercase'):
+
+        self.period = Parameter(name, 'period', 1, 1e-10, 10, tinyval)
+        self.offset = Parameter(name, 'offset', 0, 0, hard_min=0)
+        self.ampl = Parameter(name, 'ampl', 1, 1e-05, hard_min=0)
+        pars = (self.perioD, self.oFFSEt, self.AMPL)
+        self._renamedpars = [('NORM', 'ampl')]
+
+        self._basemodel = Sin()
+
+        ArithmeticModel.__init__(self, name, pars)
+
+    def calc(self, *args, **kwargs):
+        for par in self.pars:
+            setattr(self._basemodel, par.name, par.val)
+
+        self._basemodel.integrate = self.integrate
+        return self._basemodel.calc(*args, **kwargs)
+
+
+class test_model_parametercase_instance(test_model_renamed):
```

Comment:
@DougBurke I guess it is intentional that this test, as well as the one it's extending, are all running the same set of core tests, as they are run with different custom models. But I wanted to double check :)